pkg/controller/metrics/metrics.go

```diff
@@ -581,3 +589,23 @@ func GetClusterDeploymentType(obj metav1.Object) string {
 	}
 	return hivev1.DefaultClusterType
 }
+
+// ObserveControllerReconcileTime is used by all controllers in a defer block to log and report their elapsed time and
+// optionally an outcome in a consistent fashion.
+func ObserveControllerReconcileTime(controller string, start time.Time, outcome ReconcileOutcome, logger log.FieldLogger) {
```

Comment:
```suggestion
func ObserveControllerReconcileTime(controller ControllerName, start time.Time, outcome ReconcileOutcome, logger log.FieldLogger) {
```
So that all of the callers do not have to convert to a string.

Reply:
This is a really good suggestion. Noted for future PRs.